### ISSUE TITLE
[feature request] transposed convolution

### ISSUE THREAD

**Issue by Pastafarianist** [scipy/scipy#10122]
It would be convenient to have an equivalent of `tf.nn.conv2d_transpose` or `torch.nn.functional.conv_transpose2d` in Scipy. These functions are useful for the case of visualization of convolutions. This is a deep learning-specific domain, but perhaps they could be useful in other cases as well. Even in this case, using Tensorflow or PyTorch is not very convenient as they require construction and execution of a computational graph, which is an overkill for the purposes of visualization.

**Comment by rgommers** [comment]
It looks like this is the gradient of a 2D convolution, not `conv2d.transpose()`. This is probably straightforward in Tensorflow and PyTorch, because they have things like `autograd` built in to calculate gradients. I don't see how we'd be able to do this easily in SciPy. 

If I misunderstand, please explain in more detail.

**Comment by Pastafarianist** [comment]
I believe your understanding is correct. However, note that neither Tensorflow nor `autograd` in PyTorch compute the gradient of convolution automatically. For example, this is (apparently, I'm not an expert in the codebase) its implementation in Tensorflow: https://github.com/tensorflow/tensorflow/blob/0702646d5f49544a19476c791bc954f054e2b2c8/tensorflow/lite/kernels/internal/optimized/optimized_ops.h#L6353

**Comment by rgommers** [comment]
That looks nontrivial to implement and maintain. I'd like to see use cases outside of deep learning (I couldn't find any with a quick search). Otherwise I think it's out of scope.

**Comment by Pastafarianist** [comment]
I am personally not aware of any use cases outside of deep learning.

**Comment by rgommers** [comment]
Thanks @Pastafarianist. The general rule for SciPy is that we include widely applicable algorithms, if it's specific to one domain then it belongs in a domain-specific package. Also, replicating PyTorch/Tensorflow is probably not the best way to spend our fairly limited time.

So I'll close this request. If new information shows up we can always reopen it.
